Submitted By: Ryan Oliver <ryan.oliver AT pha.com.au>
Date: 2004-10-24
Initial Package Version: 0.9.7d
Origin: Ryan Oliver
Upstream Status:
Description: This patch adds a CONFIG_TARGET for x86_64 which sets
  compiler/linker flags correctly when building openssl 32bit on a bi-arch
  system. To build 32bit on bi-arch x86_64, configure with
    ./Configure linux-x86_64-32 ...

  Rediffed for 0.9.8k - By Jim Gifford 2009-03-25
  Redone for 1.0.0c - By William Harrington 2010-10-15

diff -Naur openssl-1.0.0c.orig/Configure openssl-1.0.0c/Configure
--- openssl-1.0.0c.orig/Configure	2010-11-30 22:19:26.000000000 +0000
+++ openssl-1.0.0c/Configure	2010-12-15 16:27:08.000000000 +0000
@@ -352,6 +352,7 @@
 "linux-ia64-ecc","ecc:-DL_ENDIAN -DTERMIO -O2 -Wall -no_cpprt::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHUNK DES_INT:${ia64_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-ia64-icc","icc:-DL_ENDIAN -DTERMIO -O2 -Wall -no_cpprt::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHUNK DES_RISC1 DES_INT:${ia64_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-x86_64",	"gcc:-m64 -DL_ENDIAN -DTERMIO -O3 -Wall -DMD32_REG_T=int::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHUNK DES_INT DES_UNROLL:${x86_64_asm}:elf:dlfcn:linux-shared:-fPIC:-m64:.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR):::64",
+"linux-x86_64-32",     "gcc:-m32 -DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -mcpu=k8 -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC:-m32:.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-s390x",	"gcc:-m64 -DB_ENDIAN -DTERMIO -O3 -Wall::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHAR RC4_CHUNK DES_INT DES_UNROLL:${s390x_asm}:dlfcn:linux-shared:-fPIC:-m64:.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR):::64",
 #### SPARC Linux setups
 # Ray Miller <ray.miller@computing-services.oxford.ac.uk> has patiently
